### ISSUE TITLE
BadaWithCalc Order of Arguments

### ISSUE THREAD

**Issue by rtsleight** [mitre/FMACM#19]
When ThreeDOFDynamics::Initialize(...) calls m_bada_calculator.getConfig(...), the order of arguments is:

**(cas, m_altitude_msl_at_final_waypoint, Units::MetersLength(m_dynamics_state.h)**,...). 

However, the order of arguments in BadaWithCalc::getConfig(...) is

 **(velocity_cas, altitude, altitude_faf,...)**. 

It looks like the aircraft altitude and faf altitude order is reversed. What am I missing???

**Comment by sbowman-mitre** [comment]
Hey @rtsleight , I'm unable to find this situation in the code. When I look at `ThreeDOFDynamics::Initialize()` [1], I see the correct order of parameters. I scanned all uses of `getConfig()` and couldn't find any that had reversed those inputs. Is it possible you have those modifications only in your code?

https://github.com/mitre/FMACM/blob/61622260902ae65d190672cd3f0a7d8a1fba4a34/Public/ThreeDOFDynamics.cpp#L250

**Comment by rtsleight** [comment]
Hi @sbowman-mitre, here is my interpretation of the args in [ThreeDOFDynamics::Initialize() ](https://github.com/mitre/FMACM/blob/61622260902ae65d190672cd3f0a7d8a1fba4a34/Public/ThreeDOFDynamics.cpp#L250) code referenced above:
```
   m_bada_calculator.getConfig(  
(cas)----------->         m_true_weather.getAtmosphere()->TAS2CAS(Units::MetersPerSecondSpeed(m_dynamics_state.V),(Units::MetersLength(m_dynamics_state.h)),
(faf alt)------->         m_altitude_msl_at_final_waypoint,
(aircraft alt)-->         Units::MetersLength(m_dynamics_state.h),
         0, cd0, cd2, gear, mode);
```
And the arguments for  [getConfig ](https://github.com/mitre/FMACM/blob/61622260902ae65d190672cd3f0a7d8a1fba4a34/AircraftDynamicsTestFramework/BadaWithCalc.cpp#L61) are:

```
void BadaWithCalc::getConfig(
 (cas)----------->     const Units::Speed &velocity_cas, /** [in] Calibrated airspeed in knots */
 (aircraft alt)-->     const Units::Length &altitude,    /** [in] Altitude in meters */
 (faf alt)------->     const Units::Length &altitude_faf, /** [in] Altitude at final approach fix */
      int modeLast,       /** [in] previous flap configuration (0-3) */
      double &cd0,        /** [out] parasitic drag coefficient */
      double &cd2,        /** [out] induced drag coefficient */
      double &gear,       /** [out] landing gear drag coefficient */
      int &mode           /** [out] flap configuration (0-3):
            0=cruise, 1=approach, 2=landing, 3=gear down */)
```

does that seem correct?

**Comment by sbowman-mitre** [comment]
Ah, yes, you are correct. I was miscounting my commas! Good find. We'll get out a fix soon, but you should change your code in the meantime

**Comment by sbowman-mitre** [comment]
This has been resolved with the v3.2.0 release.
